### PR TITLE
fix: change input selector to id

### LINKED ISSUE #263: Meilisearch docs glitches on refreshes
I think this is a non-intentional behavior, and it would be awesome if we could make it work without this glitch. I believe in slower connections, and the experience is not that good.

https://user-images.githubusercontent.com/4116980/163203388-6bca56fa-f545-4f42-b3c0-20e3a9d9e839.mov

### PR DESCRIPTION
**Pull request by ColinFrick**
# Pull Request

## Related issue
Fixes #263

## What does this PR do?
- Change the selector to the input id, therefor applying the styling even before the docs-searchbar.js is initialized.

This does require a change in [meilisearch/documentation](https://github.com/meilisearch/documentation) because this does not handle `@media (prefers-color-scheme: dark)`

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by bidoubiwa** [comment]
Hey @ColinFrick 
Unfortunately if it is not compatible with the dark mode we cannot add it. Do you think this is something that can be done?

**Comment by ColinFrick** [comment]
@bidoubiwa the inconsistencies result from the different types of applying the dark / light mode styling in docs-searchbar.js and the documentation CSS. docs-searchbar.js uses [data-ds-theme] to determine the theme, while the documentation page uses (prefers-color-scheme: dark) directly.

Hm, because we set the theme through the plugin, it should be possible to apply the corresponding theme on the plugin tag directly without waiting for docs-searchbar.js to be initialized.

**Comment by ColinFrick** [comment]
@bidoubiwa I have changed the PR. This should cover any combination of prefers-color-scheme / enableDarkMode, and does not require any changes in the documentation repository.

**Comment by bidoubiwa** [comment]
> docs-searchbar.js uses [data-ds-theme] to determine the theme, while the documentation page uses (prefers-color-scheme: dark) directly.

Do you think it is worth to change that in docs-searchbar.js ? As it is not exposed to the user.

**Comment by ColinFrick** [comment]
Even if we change docs-searchbar.js we still would need it here. It really comes down to that split second between rendering the DOM (with Vue component) and wrapping the component with docs-searchbar.js

ezgif-3-2ba7f0c446

File: MeiliSearchBox.vue [hunk 1]
```diff
@@ -72,9 +75,14 @@ export default {
...
 
     update(options) {
+      this.updateTheme(options)
       this.$el.innerHTML =
         '<input id="meilisearch-search-input" class="search-query">'
       this.initialize(options)
+    },
+
+    updateTheme(options) {
+      this.theme = options.enableDarkMode === true ? 'dark' : options.enableDarkMode === false ? 'light' : 'auto'
```
Comment (bidoubiwa) [inline MeiliSearchBox.vue:85]: If the value is set to `auto`  it is considered `true` and thus it will always showcase the dark mode.

I changed your ternary nesting with if/else for readability
```suggestion
    updateTheme(options) {
      if (options.enableDarkMode) { // will enter here even if it is auto ?
         this.theme = 'dark'
      }
      else if (!options.enableDarkMode) {
         this.theme = 'light'
      }
   }
```

File: MeiliSearchBox.vue [hunk 2]
```diff
@@ -72,9 +75,20 @@ export default {
...
+    },
+
+    updateTheme(options) {
+      if (options.enableDarkMode === true) {
+        this.theme = 'dark'
+      } else if (options.enableDarkMode === false) {
+        this.theme = 'light'
+      } else {
+        this.theme = 'auto'
+      }
```
Comment (bidoubiwa) [inline MeiliSearchBox.vue:91]: This is always going to be true. 
I tried `auto` and a light theme on my computer and it is still `dark` mode

Comment (bidoubiwa) [reply MeiliSearchBox.vue:91]: Mmh, with the same computer but on chrome. Let me try again on firefox

Comment (bidoubiwa) [reply MeiliSearchBox.vue:91]: My bad! I forgot to change the value of enableDarkMode 🤦

Comment (ColinFrick) [reply MeiliSearchBox.vue:91]: enableDarkMode is already 'auto' in the playground
https://github.com/meilisearch/vuepress-plugin-meilisearch/blob/5bf8db5acfd158926b187abd6733c4788c372925/playground/.vuepress/config.js#L21

Take your time, no rush 😉

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥🔥

**Comment by bidoubiwa** [comment]
bors merge

**Comment by bidoubiwa** [comment]
bors merge

**Comment by bidoubiwa** [comment]
Thanks again for contributing with Meilisearch :heart:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive a small gift from Meilisearch too, don't forget to fill [this form](https://meilisearch.typeform.com/to/idt5qugY).
